Tools/scripts/extract_param_defaults.py

```diff
@@ -67,35 +71,47 @@ def parse_arguments(args=None):
     return args
 
 
-def extract_parameter_default_values(logfile: str) -> Dict[str, float]:
+def extract_parameter_values(logfile: str, type: str = 'defaults') -> Dict[str, float]:
     """
-    Extracts the parameter default values from an ArduPilot .bin log file.
+    Extracts the parameter values from an ArduPilot .bin log file.
 
     Args:
         logfile: The path to the ArduPilot .bin log file.
+        type: The type of parameter values to extract. Can be 'defaults', 'values' or 'non_default_values'.
 
     Returns:
-        A dictionary with parameter names as keys and their default values as float.
+        A dictionary with parameter names as keys and their values as float.
     """
     try:
         mlog = mavutil.mavlink_connection(logfile)
     except Exception as e:
         raise SystemExit("Error opening the %s logfile: %s" % (logfile, str(e))) from e
-    defaults = {}
+    values = {}
     while True:
         m = mlog.recv_match(type=['PARM'])
         if m is None:
-            if not defaults:
+            if not values:
                 raise SystemExit(NO_DEFAULT_VALUES_MESSAGE)
-            return defaults
+            return values
         pname = m.Name
         if len(pname) > PARAM_NAME_MAX_LEN:
             raise SystemExit("Too long parameter name: %s" % pname)
         if not re.match(PARAM_NAME_REGEX, pname):
             raise SystemExit("Invalid parameter name %s" % pname)
         # parameter names are supposed to be unique
-        if pname not in defaults and hasattr(m, 'Default'):
-            defaults[pname] = m.Default  # the first time default is declared is enough
+        if pname in values:
+            continue
+        if type == 'defaults':
+            if hasattr(m, 'Default'):
+                values[pname] = m.Default
+        elif type == 'values':
+            if hasattr(m, 'Value'):
+                values[pname] = m.Value
+        elif type == 'non_default_values':
+            if hasattr(m, 'Value') and hasattr(m, 'Default') and m.Value != m.Default:
+                values[pname] = m.Value
+        else:
+            raise SystemExit("Invalid type %s" % type)
```

Comment:
Why not `ValueError`?

Reply:
I plan to exit and not have a cryptic backtrace